core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -230,6 +228,61 @@ oppia.factory('TopicUpdateService', [
           // Get the current change list.
           var currentChangeList = UndoRedoService.getChangeList();
           var indicesToDelete = [];
+          var oldSubtopicIdSkillIdMap = {};
+          var index = null;
+          var skillIds = subtopic.getSkillIds();
+          for (var i = 0; i < currentChangeList.length; i++) {
+            var changeDict =
+              currentChangeList[i].getBackendChangeObject();
+            if (oldSubtopicIdSkillIdMap[changeDict.skill_id] === undefined) {
```

Comment:
if (!oldSubtopicIdSkillIdMap.hasOwnProperty(changeDict.skill_id))

Although it's usually easier to read if you check for a positive, so maybe consider doing 

if (oldSubtopicIdSkillIdMap.hasOwnProperty(changeDict.skill_id))

and switching the if/else around.

Reply:
Done, changed the loop to follow the new method.